src/app/content/components/Content.tsx

```diff
@@ -18,17 +32,118 @@ const Background = styled.div`
   min-height: 100%;
 `;
 
+// tslint:disable-next-line:variable-name
+const ContentNotifications = styled(Notifications)`
+  top: ${bookBannerDesktopHeight + toolbarDesktopHeight + navDesktopHeight}rem;
+
+  @media (max-width: ${inlineDisplayBreak}) {
+    top: ${bookBannerDesktopHeight + toolbarDesktopHeight}rem;
+    ${wrapperPadding}
+  }
+`;
+
+// tslint:disable-next-line:variable-name
+const CenteredContentRow = styled(CenteredContent)`
+  min-height: 100%;
+  display: flex;
+  flex-direction: row;
+`;
+
+// tslint:disable-next-line:variable-name
+const UndoPadding = isOpenConnector(styled.div`
+  margin-right: -${theme.padding.page.desktop}rem;
+  ${theme.breakpoints.mobile(css`
+    margin: 0 -${theme.padding.page.mobile}rem;
+  `)}
+
+  ${styleWhenSidebarClosed(css`
+    margin-left: -${theme.padding.page.desktop}rem;
+    ${theme.breakpoints.mobile(css`
+      margin-left: -${theme.padding.page.mobile}rem;
+    `)}
+  `)}
+`);
+
+// tslint:disable-next-line:variable-name
+const MainContentWrapper = isOpenConnector(styled.div`
+  overflow: visible;
+  background-color: ${mainContentBackground};
+  transition: max-width ${sidebarTransitionTime}ms;
+  max-width: ${contentWrapperMaxWidth - sidebarDesktopWidth}rem;
+  ${styleWhenSidebarClosed(css`
+    max-width: ${contentWrapperMaxWidth}rem;
+    margin: 0 auto;
+  `)}
+`);
+
+// tslint:disable-next-line:variable-name
+const HideOverflowAndRedoPadding = isOpenConnector(styled.div`
+  ${wrapperPadding}
+  ${styleWhenSidebarClosed(css`
+    ${wrapperPadding}
+  `)}
+`);
+
+/*
+ * this layout is a mess for these reasons:
+ * - the navs must have the default padding inside their containers so their
+ *   backgrounds go to the edge of the window.
+ *
+ * - the content wrapper must behave the same way as the navs in order to get
+ *   the sidebar in the right place to line up with the button in the toolbar.
+ *
+ * - the white background must then have negative margin to undo the default
+ *   padding so that it can get to the edge of the screen on small windows.
```

Comment:
It seems like this will work but is it unlikely that we will need to add styling to the body (or descendants) when a component changes state in general?

If not, then these undo styles make sense. I'm just worried that we will have many Undo's sprinkled in the code eventually.


**Or:** the ToC toggle state is in redux; could the components (or the root) pull that state out and style themselves appropriately? Or is that what you are referring to in:

> ... which means that a disturbing number of things need to know when the sidebar is open/closed

Reply:
yes i was referring to the redux state, the components do pull the redux state and style themselves accordingly. i don't love having lots of styling all over the app based on the same variable, it feels disjointed to me, but in this case i couldn't centralize the sidebar open/close styles.